pkg/controllers/csi/metadata/sqlite_gorm_client.go

```diff
@@ -0,0 +1,197 @@
+		{
+			ID:      "202403041200",
+			Migrate: dataMigration,
+			Rollback: func(tx *gorm.DB) error {
+				return nil
+			},
+		},
+	}).Migrate()
+}
+
+func (conn *DBConn) CreateTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error {
+	return conn.db.WithContext(ctx).Create(tenantConfig).Error
+}
+
+func (conn *DBConn) UpdateTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error {
+	return conn.db.WithContext(ctx).Updates(tenantConfig).Error
+}
+
+func (conn *DBConn) DeleteTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error {
+	return conn.db.WithContext(ctx).Delete(tenantConfig).Error
+}
+
+func (conn *DBConn) CreateCodeModule(ctx context.Context, codeModule *CodeModule) error {
+	return conn.db.WithContext(ctx).Create(codeModule).Error
+}
+
+func (conn *DBConn) ReadCodeModuleByVersion(ctx context.Context, version string) (*CodeModule, error) {
+	var codeModule CodeModule
+
+	result := conn.db.WithContext(ctx).First(&codeModule, "version = ?", version)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return &codeModule, nil
+}
+
+func (conn *DBConn) DeleteCodeModule(ctx context.Context, codeModule *CodeModule) error {
+	return conn.db.WithContext(ctx).Delete(codeModule).Error
+}
+
+func (conn *DBConn) ReadTenantConfigByTenantUUID(ctx context.Context, uid string) (*TenantConfig, error) {
+	var tenantConfig TenantConfig
+
+	result := conn.db.WithContext(ctx).First(&tenantConfig, "tenant_uuid = ?", uid)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return &tenantConfig, nil
+}
+
+func (conn *DBConn) CreateOSMount(ctx context.Context, osMount *OSMount) error {
+	return conn.db.WithContext(ctx).Create(osMount).Error
+}
+
+func (conn *DBConn) UpdateOSMount(ctx context.Context, osMount *OSMount) error {
+	return conn.db.WithContext(ctx).Updates(osMount).Error
+}
+
+func (conn *DBConn) ReadOSMountByTenantUUID(ctx context.Context, tenantUUID string) (*OSMount, error) {
+	var osMount OSMount
+
+	result := conn.db.WithContext(ctx).Preload("VolumeMeta").First(&osMount, "tenant_uuid = ?", tenantUUID)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return &osMount, nil
+}
+
+func (conn *DBConn) DeleteOSMount(ctx context.Context, osMount *OSMount) error {
+	if osMount.VolumeMeta != (VolumeMeta{}) {
+		conn.db.WithContext(ctx).Delete(&osMount.VolumeMeta)
+	}
+
+	return conn.db.WithContext(ctx).Delete(osMount).Error
+}
+
+func (conn *DBConn) CreateAppMount(ctx context.Context, appMount *AppMount) error {
+	return conn.db.WithContext(ctx).Create(appMount).Error
+}
+
+func (conn *DBConn) UpdateAppMount(ctx context.Context, appMount *AppMount) error {
+	return conn.db.WithContext(ctx).Updates(appMount).Error
+}
+
+func (conn *DBConn) ReadAppMountByVolumeMetaID(ctx context.Context, volumeMetaID string) (*AppMount, error) {
+	var appMount AppMount
+
+	result := conn.db.WithContext(ctx).Preload("VolumeMeta").First(&appMount, "volume_meta_id = ?", volumeMetaID)
```

Comment:
same here

Reply:
That won't work as stated above: in case of invalid tenantUUID it will create an invalid object instead of returning an error- at least that's what my tests showed.

Reply:
For me, one or another is redundant; it's unclear why do we pass `volumeMetaID` 2 times.

Reply:
fixed in #1fd425db51879340cc674fbeb913a9eb5619763b